### PR TITLE
bugfix: fix that ContainerManager.Create wouldn't remove snapshot when failed

### LINKED ISSUE #2061: bugfix: ContainerManager create haven't deal with error
### Ⅰ. Issue Description
ContainerManager Create haven't deal with error, no set defer to err != nil, such as
succeed to `CreateSnapshot`, but do `generateMountPoints` is error, so just return
and Snapshot is remain. 

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1.
2.
3.

### Ⅵ. Environment:

- pouch version (use `pouch version`):
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by xiaoxubeii**
### Ⅰ. Describe what this PR did
Fix that ContainerManager.Create wouldn't remove snapshot when failed.

### Ⅱ. Does this pull request fix one issue?

fixes #2061.

### Ⅲ. Describe how you did it
Add a variable cleanups to hold cleanup function, and loop it when failed in defer.

### CODE REVIEW

**Comment by allencloud** [comment]
Thanks a lot for your contribution. Just double check could we add an integration test for this case? @xiaoxubeii

**Comment by xiaoxubeii** [comment]
@allencloud OK, i will add a new integration testcase.

**Comment by xiaoxubeii** [comment]
@allencloud I find it is a little difficult to add a integration testcase of snapshot cleanup. That is because there is no api interface to list snapshots and i also cannot access containerd directly because of lacking the configuration. Any idea can help me to test?

File: daemon/mgr/container.go [hunk 1]
```diff
@@ -303,6 +314,9 @@ func (mgr *ContainerManager) Create(ctx context.Context, name string, config *ty
 	if err := mgr.Client.CreateSnapshot(ctx, id, config.Image); err != nil {
 		return nil, err
 	}
+	cleanups = append(cleanups, func() {
+		mgr.Client.RemoveSnapshot(ctx, id)
```
Comment (fuweid) [inline daemon/mgr/container.go]: could we add some log here?

**Comment by allencloud** [comment]
LGTM

**Comment by allencloud** [comment]
Thanks for your great work. 🍺 @xiaoxubeii

**Pull request merged**